src/js/select2/core.js

```diff
@@ -206,6 +206,23 @@ define([
       childList: true,
       subtree: false
     });
+
+    var ancestors = this.$element.parents('fieldset');
```

Comment:
Only concern here is that we only try to figure this out _once_, which works for 90% of use cases but fails for a common one: dynamically added elements. It's common for Select2 to be initialized on an element before that element is added to the DOM.

I'm not sure of a great way for us to detect that and properly apply the observer.

Reply:
I actually considered that while creating the original patch, but it was 5am and the only solution I could think of was "attach an observer to the document root and monitor subtree changes", but that would obviously have enormous performance implications.

I'll give it some more thought, but honestly every moment feels like 5am these days...

Reply:
> I'll give it some more thought, but honestly every moment feels like 5am these days...

I 100% feel that. Not expecting any urgency on a fix for this or for you to be the one to provide it. We're all in this together.

---

I think this can be done with another set of mutation observers, but I would need to look more into it. It's tricky because most of the docs on things like this are for how to detect nodes being inserted into the current element, not for how to detect when the current element is inserted into the DOM. I'm sure we'll find an answer other there.